### PR TITLE
Add details of Jaeger binary architecture

### PR DESCRIPTION
**Pull request by jkowall**
## Which problem is this PR solving?
Part of #777 

## Description of the changes
Added new diagram
Provided details of all components, ref:
- https://github.com/jaegertracing/jaeger/blob/5475a36170f5393fd4332858108b16d465c56551/cmd/jaeger/config.yaml#L2
- https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/components.go
- https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal

## How was this change tested?
local make develop 

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits

### CODE REVIEW

File: content/docs/next-release-v2/architecture.md [hunk 1]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
+
+#### Recievers
+* [Jaeger](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jaegerreceiver)
+
+* [Kafka}](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkareceiver)
+
+* [Zipkin](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/zipkinreceiver)
+
+* [OTLP](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver)	
+
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:88]:
```suggestion
```

File: content/docs/next-release-v2/architecture.md [hunk 2]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
+## Jaeger binary components
+
+The Jaeger binary consists of several components, some of which are derived from OpenTelemetry and some are internal. The following diagram has the major components.
+
+![Architecture](/img/architecture-v2-binary.png)
+
+Aside from these components there are sevral other components from OpenTelemetry you can use in the config of the Jaeger binary. Here is the full list of components:
+
+### Jaeger Components
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:67]: I am not sure we should be linking to the source code for these. We already have some documentation on the Configuration page https://deploy-preview-783--romantic-neumann-1959d7.netlify.app/docs/next-release-v2/configuration/

File: content/docs/next-release-v2/architecture.md [hunk 3]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
+
+* [Adaptive Sampling Processor](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/processors/adaptivesampling)
+
+* [Jaeger Storage Exporter](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/extension/jaegerstorage)
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:71]: also `jaeger_storage` extension, which is actually the 90% of Jaeger and define all storage backends. I would put it first on the list.

File: content/docs/next-release-v2/architecture.md [hunk 4]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
+
+* [Jaeger Query Extension](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/extension/jaegerquery)
+
+* [Jaeger Remote Sampling Extension](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/extension/remotesampling)
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:75]: group adaptive sampling and remote sampling close together

File: content/docs/next-release-v2/architecture.md [hunk 5]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
\ No newline at end of file
+* An extra layer of marshaling/unmarshaling the data.
+
+## Jaeger binary components
+
+The Jaeger binary consists of several components, some of which are derived from OpenTelemetry and some are internal. The following diagram has the major components.
+
+![Architecture](/img/architecture-v2-binary.png)
+
+Aside from these components there are sevral other components from OpenTelemetry you can use in the config of the Jaeger binary. Here is the full list of components:
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:65]:
```suggestion
Aside from these components there are several other components from OpenTelemetry you can use in the config of the Jaeger binary. Here is the full list of components:
```

File: content/docs/next-release-v2/architecture.md [hunk 6]
```diff
@@ -54,4 +54,64 @@ Benefits:
...
+#### Recievers
+* [Jaeger](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jaegerreceiver)
+
+* [Kafka}](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkareceiver)
+
+* [Zipkin](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/zipkinreceiver)
+
+* [OTLP](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver)	
+
+#### Processors
```
Comment (yurishkuro) [inline content/docs/next-release-v2/architecture.md:89]: would be useful to add a sentence to each explaining what they are for.

**Review by yurishkuro — COMMENTED** [review]
It doesn't fully address the ticket, as I would like the diagram to be more detailed w.r.t. the core extensions

**Comment by jkowall** [comment]
> It doesn't fully address the ticket, as I would like the diagram to be more detailed w.r.t. the core extensions

Can you provide what you mean by "core extensions" that you'd like to see?

**Comment by yurishkuro** [comment]
> Can you provide what you mean by "core extensions" that you'd like to see?

`jaeger_storage`, `jaeger_query` and `jaeger_storage_exporter` are what I consider core components, since without them there is no Jaeger-v2.

File: content/docs/next-release-v2/architecture.md (hunk 2 above)
Comment (jkowall) [reply content/docs/next-release-v2/architecture.md:67]: I still think it's good for users to get the context of the binary and where the sources come from.

**Comment by jkowall** [comment]
Add of these should be addressed with the latest push (two).

**Comment by yurishkuro** [comment]
linter complains of missleppings
```
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:79:6 - Unknown word (Recievers) fix: (Receivers)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:80:103 - Unknown word (Recieve) fix: (Receive)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:82:115 - Unknown word (Recieve) fix: (Receive)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:84:113 - Unknown word (Recieve) fix: (Receive)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:86:115 - Unknown word (Recieve) fix: (Receive)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:98:126 - Unknown word (OLTP)
/home/runner/work/documentation/documentation/content/docs/next-release-v2/architecture.md:100:1[18](https://github.com/jaegertracing/documentation/actions/runs/11921650283/job/33226240441?pr=783#step:4:19) - Unknown word (OLTP)
```

**Comment by jkowall** [comment]
Should be fixed now, sorry for the missed spelling correction.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**